Sprint #3

sineergin opened the pull request:
New features in this pull request include:

- An Edit-profile-page: Where the logged in user can change her interests and type in her country and whatsapp number, which will be displayed on her profile.

- A Notification-page:  Where the logged in user can see the status of bumps sent.

sineergin committed:
create an edit page where currentUser can change her interests and also change the country she wishes displayed on her profile

sineergin committed:
fix profile-section to show either "People who also like <interest>" or "People who like <interest>", depending on if currentUser likes the same nterests or not. I

sineergin committed:
clean up code

carolinebjorner committed:
create notification pages and components with mock data for now

Thiberg committed:
Irrelevant file removal

Thiberg committed:
Buttons.jsx fix

Thiberg committed:
Buttons.jsx fix

Thiberg committed:
No more duplicate bumps, and the user is notified about it.

Thiberg committed:
No more duplicate bumps, and the user is notified about it.

Thiberg committed:
Merge pull request #32 from TID-A25/bump+button_fixes

Buttons & Bump page fixes implementation

sineergin committed:
Merge remote-tracking branch 'origin/development' into fix-common-interests

vicvar01 committed:
Merge pull request #33 from TID-A25/fix-common-interests

Fix common interests logic between currentUser and otherusers

sineergin committed:
Merge branch 'development' into edit-profile-page

vicvar01 committed:
Merge branch 'development' into edit-profile-page

vicvar01 committed:
Merge pull request #34 from TID-A25/edit-profile-page

Create edit-page for currentUser to change her profile settings

vicvar01 committed:
Merge remote-tracking branch 'origin/edit-profile-page' into HEAD

vicvar01 committed:
Merge branch 'development' into HEAD

Thiberg committed:
Fixed settings not showing up error

vicvar01 committed:
Merge pull request #36 from TID-A25/merge_conflicts

Resolved merge conflicts

carolinebjorner committed:
Merge branch 'development' into notifications-page

carolinebjorner committed:
Merge pull request #37 from TID-A25/notifications-page

Start creating notification page and component

Thiberg committed:
Parsing logic from 'common_interests' branch

Thiberg committed:
Styling changes from Sine Ergin

vicvar01 committed:
Merge pull request #38 from TID-A25/merger_err_fixing

Final merge error fixes

carolinebjorner committed:
Remove Profiles button from Header

sineergin committed:
Add Whatsapp section to profile info and profile edit page

vicvar01 committed:
Merge pull request #39 from TID-A25/add-whatsapp-section

Add Whatsapp section to profile info and profile edit page

carolinebjorner committed:
add welcome message to Home page

carolinebjorner committed:
Merge remote-tracking branch 'origin/development' into Home-feed

vicvar01 committed:
Merge pull request #40 from TID-A25/Home-feed

Home-feed updates

vicvar01 committed:
adding backend logic to notifications + hooks and queries

vicvar01 committed:
adding refresh  icon instead of button

carolinebjorner committed:
Refactor the first fields into a component

vicvar01 committed:
removing duplicate code from notification list

carolinebjorner committed:
minor styling changes

vicvar01 committed:
editing my comment -,-

vicvar01 committed:
Merge remote-tracking branch 'origin/development' into notification-page-backend

carolinebjorner committed:
Merge pull request #41 from TID-A25/notification-page-backend

connecting notification page to backend

carolinebjorner committed:
Merge branch 'development' into edit-profile-styling

vicvar01 committed:
Merge pull request #42 from TID-A25/edit-profile-styling

minor refactor and styling of edit profile

vicvar01 committed:
fixing overlapping css for profile page and edit profile page

vicvar01 committed:
Merge branch 'development' into fixin-image-css-userprofile

carolinebjorner committed:
Merge pull request #43 from TID-A25/fixin-image-css-userprofile

css fixes for interest images

vicvar01 committed:
moving currentuserid to a file instead of hardcoding everywhere

Thiberg committed:
N+1 query fix

Thiberg committed:
Visible indication of bump sent

Thiberg committed:
Changed visual of profile you've already bumped to remove bump option

Thiberg committed:
Comments

vicvar01 committed:
adding more profiles, userswitcher to switch between them, and updating hooks

vicvar01 committed:
changing settings icon to edit icon

carolinebjorner committed:
add simple navigation to buttons on BumpSent

Thiberg committed:
Removed parsing and hooks from pages

vicvar01 committed:
Merge pull request #45 from TID-A25/siwtch-between-user-logins

Adding a possibility to switch between profiles for demo purposes

Thiberg committed:
Merge remote-tracking branch 'origin/ready_for_dev' into jtt_S3

Thiberg committed:
Merge pull request #46 from TID-A25/jtt_S3

Fixing parsing & hooks, and added "Waiting for response"

carolinebjorner committed:
add link to otherUser profile photo + fix photo visibility

carolinebjorner committed:
add copy over cancel button

carolinebjorner committed:
add toast when cancel is clicked

carolinebjorner committed:
Merge remote-tracking branch 'origin/ready_for_dev' into bump-sent-page

Thiberg committed:
Added Carolines atrium back to the Home page

carolinebjorner committed:
Merge pull request #47 from TID-A25/bump-sent-page

Bump sent page updates

Thiberg committed:
Merge remote-tracking branch 'origin/ready_for_dev' into atrium-recovery

carolinebjorner committed:
Merge pull request #48 from TID-A25/atrium-recovery

Added Carolines atrium back to the Home page

Thiberg committed:
Interest title recovered

louiselynggaard commented on src/App.jsx:
```diff
@@ -1,13 +1,15 @@
 // ** importing Parse as configured in vite.config.js:
-import "./constants/parseConfig.js"
+import "./constants/parseConfig.js";
 import { BrowserRouter, Routes, Route, Link } from "react-router-dom";
```
You can remove `Link` here when you are no longer using it

vicvar01 replied:
unused imports fixed

louiselynggaard commented on src/App.jsx:
```diff
@@ -18,8 +20,10 @@ function App() {
           <Route path="/bump-sent/:otherUserId" element={<BumpSent />} />
           <Route path="/bump-received" element={<BumpReceived />} />
+          <Route path="/notifications" element={<Notifications />} />
         </Routes>
```
Consider adding a fallback route so users don’t just see a blank page on unknown URLs. `path="*" element={<NotFound />}` and then do the `NotFound` component (can be very basic)

vicvar01 replied:
this is fixed

louiselynggaard commented on src/components/buttons/ActionButtons.jsx:
```diff
@@ -0,0 +1,31 @@
+  return <SingleButton size={size} color={color} label={label} onClick={onClick} />;
+}
+
+export { BackButton, SingleButton, BumpButtons };
```
nice!!

louiselynggaard commented on src/components/buttons/BumpButtons.jsx:
```diff
@@ -0,0 +1,31 @@
+  // default action: go back to home if no handler provided
+  const navigateHome = () => navigate("/");
+  const primaryHandler = onClick ? onClick : navigateHome;
+  const secondaryHandler = onSecondaryClick ? onSecondaryClick : onClick ? onClick : navigateHome;
```
this is a bit off. the Cancel/Decline button will use its own handler only if provided, otherwise fall back to the same handler as the big button.
primaryHandler = onClick
secondaryHandler also = onClick because the first branch onSecondaryClick is false, the second branch onClick is true
so, both buttons end up doing the same action?

carolinebjorner replied:
fixed

louiselynggaard commented on src/components/buttons/ActionButtons.jsx:
```diff
@@ -0,0 +1,31 @@
+ * Compatibility wrapper around focused button components.
+ * Keeps the original `Buttons` API but delegates rendering.
+ */
+export default function ActionButtons({
```
really nice that you implemented this! 

buuut
the wrapper accepts many props (size, color, label, etc.) but only forwards some of them depending on mode. In bump and back modes, most props are ignored. This leads to confusing behavior and makes the API a bit unpredictable.
consider forwarding all props and let the underlying components decide what they use

louiselynggaard replied:
it would be something like 
`export default function ActionButtons(props) {
  const { mode = "single" } = props;
  if (mode === "bump") return <BumpButtons {...props} />;
  ...`

carolinebjorner replied:
fixed

louiselynggaard commented on src/components/editInformation/EditInformation.jsx:
```diff
@@ -0,0 +1,53 @@
+        <div className="col-stack">
+          <label>
+            WhatsApp number
+            <input
```
consider adding a type here so it does not default to `text`. then I wouldn't be able to have letters in my number

sineergin replied:
Done

louiselynggaard commented on src/components/footer/Footer.jsx:
```diff
@@ -16,21 +16,20 @@ export default function Footer() {
   const [active, setActive] = React.useState("home");
```
because `Footer` re-mounts on every route change, the state always resets to "home" when the user navigates. if you go to your `/user-profile` and refresh the page, you will see that the `Footer` thinks you are on `/home`. consider using `useLocation()` instead

vicvar01 replied:
fixed

louiselynggaard commented on src/components/notifications/NotificationItem.jsx:
```diff
@@ -0,0 +1,42 @@
+      message = `Your bump has been accepted. You can now see their Whatsapp`;
+      break;
+    case "accepted_by_current_user":
+      message = `You accepted ${actorName}'s bump, now you can see their whatsapp`;
```
a small detail, but you use `whatsapp`, `WhatsApp`, `Whatsapp`. Consider doing the same all over

sineergin replied:
Done

louiselynggaard commented on src/components/profileSection/ProfileSection.css:
```diff
@@ -1,5 +1,20 @@
+  font-weight: 400;
+}
+
+/* ============================================
```
remember to remove these comments as they can look a bit like a robot made them

vicvar01 replied:
fixed

louiselynggaard commented on src/pages/EditProfile.jsx:
```diff
@@ -0,0 +1,210 @@
+import "./Pages.css";
+
+export default function EditProfile() {
+  const CURRENT_USER_ID = "C6YoifVWmr";
```
for your final, this should come from `Parse.User.current()` or a prop, not a constant
will make more sense when you can sign up and log in

vicvar01 replied:
we are not going to do login. :D:D we have something for demo purposes

louiselynggaard commented on src/pages/EditProfile.jsx:
```diff
@@ -0,0 +1,210 @@
+  };
+
+  // Save country and sync interests
+  async function handleSave(e) {
```
N+1!!
right now, when you add each new interest, you send a new query to the backend to find that interest by name. if a user selects 10 interests, you make 10 separate database queries, but you already have all interests stored in `allInterests`. This list includes both the display name and the Parse object for each interest, so you can look up the interest locally in memory, instead of asking the server again

sineergin replied:
done

vicvar01 replied:
this has been fixed

louiselynggaard commented on src/hooks/useProfiles.js:
generally, it looks good, but since this hook is only used by the Home page, it might be doing a bit too much. It’s fetching data AND shaping it into `profilesByInterest`, which is very specific to this one view.
It is fine to do so, but be prepared to defend this. You could also either inline the effect + state directly into Home, or keep a smaller hook that only returns { profiles, currentUserInterests, loading, error, refresh } and let Home do the grouping by interest. That keeps the data-fetching logic simple and makes it clearer that this grouping is a UI concern specific to the Home page

vicvar01 replied:
fixed

louiselynggaard commented on src/hooks/useNotifications.js:
This hook is nicely focused, as it only fetches notifications, manages state, exposes refresh. See comment on `useProfiles`

louiselynggaard commented on src/services/parseQueries.js:
```diff
@@ -1,137 +1,269 @@
+    if (excludeUserId) {
+      userQuery.notEqualTo("objectId", excludeUserId);
+    }
+
```
again a N+1 query pattern: 1 query for users, then 1 query per user for their interests --> slow with bigger datasets

louiselynggaard replied:
also, you are duplicating mapping logic with `fetchProfileById`, so maybe consider moving this logic elsewhere and using it in both cases

vicvar01 replied:
fixed

louiselynggaard commented on src/services/parseQueries.js:
```diff
@@ -1,137 +1,269 @@
+  }
 }
 
 export async function fetchProfileById(id) {
```
also some redundancy here. the logic is almost copy-pasted from `fetchProfiles` inner mapping

vicvar01 replied:
fixed

louiselynggaard commented on src/services/parseQueries.js:
```diff
@@ -1,137 +1,269 @@
+}
+
+/*Get notifications (bumps) for a specific user*/
+export async function fetchNotifications(userId) {
```
Here you rely on `requestedBy` always being set and do `requestedBy.id` directly. If the field is missing or null on older records, this will throw an error. Adding a null-check around `requestedBy` before using `requestedBy.id` would make this more stable

vicvar01 replied:
fixed

louiselynggaard reviewed (changes requested):
nice work! left a few comments

vicvar01 committed:
remmoving hook from bumpsent page

carolinebjorner committed:
Merge pull request #49 from TID-A25/fix-bumpsent-page-error

remmoving hook from bumpsent page

carolinebjorner committed:
Merge pull request #50 from TID-A25/ready_for_dev

Merging ready_for_dev dev version into development

vicvar01 committed:
adding nullcheck in parsequeries

vicvar01 committed:
removed import

vicvar01 committed:
changing css comment

carolinebjorner committed:
change footer buttons to fill according to page

sineergin committed:
fix WhatsApp naming and make the type = number

carolinebjorner committed:
Merge pull request #51 from TID-A25/fix_whatsapp

fix WhatsApp naming and make the type = number

carolinebjorner committed:
Merge pull request #52 from TID-A25/footer

change footer buttons to fill according to page

vicvar01 committed:
creating a 404 page

vicvar01 committed:
Merge pull request #53 from TID-A25/nullcheck-parsequeries

Fixed nullcheck in parsequeries.js, added a 404 page + other minor stuff

carolinebjorner committed:
add secondaryHandler to BumpButtons

carolinebjorner committed:
pass through props from singleButton to ActionButtons

carolinebjorner committed:
small change to label on cancel button

vicvar01 committed:
simplifying hook and moving grouping logic to the home.jsx instead

sineergin committed:
Merge pull request #54 from TID-A25/buttons

SecondaryHandler and passing through props

vicvar01 committed:
Merge branch 'development' into simplify-useprofiles-hook

carolinebjorner committed:
edit copy on bumpHeader

vicvar01 committed:
Merge pull request #56 from TID-A25/bumpsent-text

edit copy on bumpHeader

vicvar01 committed:
Merge pull request #55 from TID-A25/simplify-useprofiles-hook

simplified the useprofiles hook

vicvar01 committed:
removing unusued assets + line

vicvar01 committed:
removing unused imports part 2

Thiberg committed:
Removed redundancies

vicvar01 committed:
Merge pull request #58 from TID-A25/redundancies

Removed redundancies

carolinebjorner committed:
Merge pull request #59 from TID-A25/removing-unused-stuff-in-files

removing unused assets and imports

vicvar01 commented:
@louiselynggaard Thanks for the feedback! We have addressed all of your comments. Pls merge B-)

vicvar01 committed:
Merge branch 'development' into bump-accept-page

vicvar01 committed:
redoing the bump accept page

vicvar01 committed:
changing copy and css for bumpstates

vicvar01 committed:
created bump-accepted page and made option to delete bump

vicvar01 committed:
allowing to cancel bump in bumpreceived

sineergin committed:
Merge pull request #60 from TID-A25/bump-accept-page

made a bump accept + bump received page

sineergin committed:
Implement whatsapp phone_visibility logic to allign with bump-status logic.

WhatsApp shows only if it’s your own profile, if the owner set visibility to all, or it’s set to bumps and the bump status is accepted. The default is set to “bumps”, meaning people who are bumped can see each others whatsapp number.

sineergin committed:
fix input field

carolinebjorner committed:
Merge pull request #61 from TID-A25/whatsapp_logic

Whatsapp-logic implemented

sineergin committed:
change from alert to toast, and change color of toast

vicvar01 committed:
Merge pull request #62 from TID-A25/toast-instead-of-alert

change from alert to toast, and change color of toast

vicvar01 committed:
removing line giving extra margin

vicvar01 committed:
removing mock profile data as it's no longer needed

vicvar01 committed:
Merge pull request #63 from TID-A25/fixing-page-layout-for-different-users

fix padding on userprofile

Thiberg committed:
Updated to work with Cloud code

sineergin committed:
change button text to fit button action. Went from "Or go back home" to "Or go back"

sineergin committed:
fix one more button

vicvar01 committed:
Merge pull request #64 from TID-A25/fix-button-text

Change button text

vicvar01 committed:
adding shared interest message to differentiate between error msgs in interests

vicvar01 committed:
adding padding to interests to userprfoile

Thiberg committed:
Change user added to main.js in back4app cloud code

Thiberg committed:
Merge pull request #65 from TID-A25/fixing-interests-when-bumped

fix error msg in interests when bumped vs when no interests

vicvar01 committed:
Merge pull request #66 from TID-A25/b4a-cloud-function

Secure profile ID's

Thiberg committed:
Colour blind friendly

carolinebjorner committed:
Merge pull request #67 from TID-A25/colours

Colour blind friendly

Thiberg committed:
Changed loading to be in middle of the screen to avoid it being hidden

carolinebjorner committed:
Merge pull request #68 from TID-A25/Loading-visibility

Changed loading to be in middle of the screen to avoid it being hidden

Thiberg committed:
README

vicvar01 committed:
Merge pull request #69 from TID-A25/README

README

Thiberg committed:
README updated

vicvar01 committed:
Merge pull request #70 from TID-A25/README

README
